core/src/main/scala/org/apache/spark/serializer/SerializationDebugger.scala

```diff
@@ -220,6 +255,27 @@ private[spark] object SerializationDebugger extends Logging {
     override def writeByte(i: Int): Unit = {}
   }
 
+  /** An output stream that emulates /dev/null */
+  private class NullOutputStream extends OutputStream {
+    override def write(b: Int) { }
+  }
+
+  /**
+   * A dummy [[ObjectOutputStream]] that saves the list of objects written to it and returns
+   * them through `outputArray`.
```

Comment:
can you expand on this a little? e.g. why pass it null output stream, what is it for, why we need to extend `ObjectOutputStream` etc.